### ISSUE TITLE
Examples: Iteration Proposal on Accessibility

### ISSUE THREAD

**Issue by thygrrr** [raylib-cs/raylib-cs#250]
- [x] The issue is specific to Raylib-cs and not raylib (?)
- [x] I checked there is no similar issue already reported

### Issue description

Some example screens violate accessibility principles, which reduces both the visual appeal and the usability of the examples:  
*(all of these apply "sometimes")*
- very small text
- low contrast text
- camera looking in wrong direction
- no alternate controls to mouse controls
- cursor keys, but not WASD
- no interaction prompt at all
- missing / concealed / elided / wrapped text
- no prompt how to calibrate mouselook
- no mouselook auto-calibration
- no explanation of what is happening on the screen
- there's no example selector (other than command line argument)
- can't move back to previous example

### Environment

Windows 11 Pro, probably elsewhere, too.

### Issue screenshot (non-exhaustive list)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/0c0c677d-838b-4174-a9fc-4df1a0eb79a3)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/757f0e5a-645e-488c-b92d-8e723239af69)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/ba29cf67-385a-4a74-815e-fb4771ad00c3)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/8be8bf8e-07c6-44d4-853c-7f2127acbed4)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/5d181d63-7b65-44fd-be55-09ef78496b01)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/6778ec5f-be8d-4c45-b59a-bab923ecc01b)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/814d634b-ce50-41a2-9312-23c99c9b597d)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/b2358177-84da-41af-a1cb-b9aabb8ea66e)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/3faf4d67-c076-4a52-8fc5-b4d6084030e9)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/6918db1e-970e-4046-a338-d2e3215742c3)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/ba6f7748-c6cd-44a9-8f81-ad7e11d807fd)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/54f615a4-f457-4a67-84f7-80a0fd3f7c8e)

![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/c6015581-fa91-4d99-a09a-d2f6fec9847f)

### Code example

I would like to discuss a unifying pass across the examples:
- unify screen layouts (where the prompts are, how they are worded, indented, etc.)
- unify fonts used (where applicable) and possibly use more legible font (sacrilege? :))
- select and apply a triadic (or split complementary color) palette for texts and interactive elements (ideally from raylib builtin colors)
- palette should also include a set of lowlights / dimmed / disabled colors
- aim for 5:1 contrast (or better) in all text
- add dark boxes or outlines around bright text that renders on dark and bright or noisy images
- increase minimum text size across the board
- add short summary explanations text to each example screen
- add clear interaction prompts with a unified control scheme
- everything should be controllable with keyboard only (mouse optional)
- auto-calibrate mouselook (it's somehow very broken on Windows 11, may warrant a separate issue)
- create a simple example selection screen (that comes back up when leaving examples)
- create a simple baseline GUI to switch between examples that is available everywhere

Each individual one of these is simple, but the sheer number of examples and differences in taste warrant a detailed discussion and action plan before submitting a +9000/-9000 line PR 🗡️ 

Also, I don't know whether this is within the scope of this repository.

**Comment by thygrrr** [comment]
Tbh, the contrast issue seems to be an overall narrative in Raylib. 🫨 
![image](https://github.com/chrisdill/Raylib-cs/assets/8904620/c44e5683-ef96-4f10-9257-14c32449d2d8)

**Comment by chrisdill** [comment]
@thygrrr Definitely in scope though needs planning like you said. Not against large PR's if necessary though splitting it up makes review easier. :)

For context the examples were copied from raylib then ported across so initially the design was to make that as easy for me as possible. For example the use of `using static Raylib_cs.Raylib;` cause I didn't want to go through and type Raylib. everywhere. Not ideal but it helped get it going quickly.

Over time I have made some changes though mostly styling, naming and porting new examples from raylib. I have wanted to do more with it but never got around to it due to the work involved. At one point I tried a imgui demo as I also wanted an example selection screen. Could revisit that idea using https://github.com/raylib-extras/rlImGui-cs prehaps.

Accessibility wise feel free to update the example designs. As long as the example fulfills the same general purpose then that works for me.

**Comment by thygrrr** [comment]
Woohoo!
